### ISSUE TITLE
UX Problems: Root Password Dialog in SLE Micro Installation

### ISSUE THREAD

**Issue by shundhammer** [yast/yast-installation#966]
_(In this repo so the UX issues are all near each other)_

From

https://documentation.suse.com/sle-micro/5.0/single-html/SLE-Micro-installation/#article-installation

![install_authentication_slem](https://user-images.githubusercontent.com/11538225/121172615-2ea25580-c858-11eb-94c0-19ca7751799f.png)

**Comment by shundhammer** [comment]
BTW is there really no way to undo selecting an imported SSH key? This should probably also go to the list of options ("_None_" ?).

**Comment by dgdavid** [comment]
Let's try to involve @kwwii here too :)

**Comment by ancorgs** [comment]
I will not say we should bring the help panel in its original form. But I cannot help thinking we have quite some space at the left, below the title, that could be used to provide a bit of context and also to serve as some kind of "invitation" to use the full help if that info is not enough.

Something along this line (this is a very crude edit done with Gimp in 2 minutes):

![draft-help](https://user-images.githubusercontent.com/3638289/121181060-e1c37c80-c861-11eb-8a23-32fa05e8c41b.png)

**Comment by joseivanlopez** [comment]
It is crazily amazing how many things can be improved in such a simple dialog like this! I agree, the title is confusing instead of helping. I infinitely prefer the shorted version proposed by @shundhammer. And I also like a lot the preliminary help with the button down as proposed by @ancorgs. I only wonder how to place it in the installer in case that other dialog layouts are used (e.g., in TW). And regarding the buttons for importing ssh keys, totally agree again, there are much better ways to do it. Personally, I prefer a popup more than a menu button. The popup allows more possibilities, like visible short help, etc. I would like something like:

~~~
Import Public SSH Key 
  imported from /dev/sda1  [Import]
~~~

**Comment by joseivanlopez** [comment]
Moreover, having the *Test Keyboard Layout* widget always visible could be questionable. This forces the user to always go through it when moving with tabs. I would prefer to offer it as an optional button or something like that.

**Comment by shundhammer** [comment]
I am torn about this keyboard test field. On one hand, it distracts from the main purpose of this dialog; it feels out of place. Anybody not familiar with the unfortunate variations of keyboard layouts (in the German layout `Z` and `Y` are exchanged with regards to the US layout: `QUERTZ` vs. `QUERTY`) will be confused.

On the other hand, if a user has such a very different keyboard layout and it's not active while setting the password, the next login will be a nasty surprise.

Actually, what I would really prefer is the Android way of (at least optionally) showing each letter for a short while; a second or two. That gives some confidence what was entered. Yes, it's not as safe against somebody looking over your shoulder as that "show only a black circle" thing that Qt does by default.

I just checked, and for a moment I thought that QLineEdit now supports that Android "echo for a moment" mode: https://doc.qt.io/qt-5/qlineedit.html#EchoMode-enum

...but it's not like that: `QLineEdit::PasswordEchoOnEdit` shows the _complete_ text whenever the widget has the keyboard focus, and when you move the focus to another widget, you only see a black filled circle for each character. That's not what we need here, and I am pretty sure our security people are not going to like that.

**Comment by hellcp** [comment]
You could always offer switching keyboard layout on every screen, for that matter having accessibility settings dialog that is accessible from every screen would be pretty helpful

**Comment by shundhammer** [comment]
Making users aware of the problem is the challenge here. Most users won't even notice that a key they are using for a password delivers a different character than they think it does.

Again, that goes in the general direction of doing the Android-like echo for password fields (which includes the somebody-looking-over-your-shoulder problem).

**Comment by joseivanlopez** [comment]
Well, if there is somebody looking over your shoulders, then there is a chance for him/her to see what keys you are stroking anyway. I also like the option for quickly echoing the characters. And another possible option is to temporary show the password. I have seen it in some places, where you have to keep pushed an eye icon to see the entered password. The password is automatically hidden again when you release the key.

**Comment by shundhammer** [comment]
Yes, that would be my choice as well.  Qt doesn't offer that, though. But I think with some fiddling we could make our own that behaves just like that. It's Qt, after all; you can do everything in Qt. ;-)
